include/aspect/boundary_velocity/interface.h

```diff
@@ -119,98 +109,223 @@ namespace aspect
+         * to combine them.
+         */
+        Tensor<1,dim>
+        boundary_velocity (const types::boundary_id boundary_indicator,
+                           const Point<dim> &position) const;
 
-    /**
-     * Register a velocity boundary conditions model so that it can be
-     * selected from the parameter file.
-     *
-     * @param name A string that identifies the velocity boundary conditions
-     * model
-     * @param description A text description of what this model does and that
-     * will be listed in the documentation of the parameter file.
-     * @param declare_parameters_function A pointer to a function that can be
-     * used to declare the parameters that this velocity boundary conditions
-     * model wants to read from input files.
-     * @param factory_function A pointer to a function that can create an
-     * object of this velocity boundary conditions model.
-     *
-     * @ingroup BoundaryVelocities
-     */
-    template <int dim>
-    void
-    register_boundary_velocity (const std::string &name,
-                                const std::string &description,
-                                void (*declare_parameters_function) (ParameterHandler &),
-                                Interface<dim> *(*factory_function) ());
+        /**
+         * A function that is used to register boundary velocity objects in such
+         * a way that the Manager can deal with all of them without having to
+         * know them by name. This allows the files in which individual
+         * plugins are implemented to register these plugins, rather than also
+         * having to modify the Manager class by adding the new boundary
+         * velocity plugin class.
+         *
+         * @param name A string that identifies the boundary velocity model
+         * @param description A text description of what this model does and that
+         * will be listed in the documentation of the parameter file.
+         * @param declare_parameters_function A pointer to a function that can be
+         * used to declare the parameters that this boundary velocity model
+         * wants to read from input files.
+         * @param factory_function A pointer to a function that can create an
+         * object of this boundary velocity model.
+         */
+        static
+        void
+        register_boundary_velocity (const std::string &name,
+                                    const std::string &description,
+                                    void (*declare_parameters_function) (ParameterHandler &),
+                                    Interface<dim> *(*factory_function) ());
+
+
+        /**
+         * Return a list of names of all boundary velocity models currently
+         * used in the computation, as specified in the input file.
+         */
+        const std::map<types::boundary_id, std::pair<std::string,std::vector<std::string> > > &
+        get_active_boundary_velocity_names () const;
```

Comment:
Can you describe what the return type of this function is? It looks complicated and could use some explanation :-)